libs/shared/providers/src/txButton/components/TxButton.tsx

```diff
@@ -39,9 +39,10 @@ export type TxButtonProps<
   > = ContractFunctionArgs<abi, 'nonpayable' | 'payable', functionName>,
 > = {
   label?: string;
+  insufficientGasLabel?: string;
```

Comment:
So ever user of this component is going to have to code this logic? Doesn't that seem wasteful?

Reply:
I think I need to add context for you to understand why I added it. I can't even click the button here because simulateData doesn't exist. The button does not work and there is no feedback as to why that is happening. We don't even have the potential to go into our wallet and see we don't have enough funds/gas.

So every user of this component will need to handle a failure like this in their onSimulateError callback. Every contract write might fail because of this. So we'll spread that logic throughout the entire app? I need convincing

Reply:
This component purpose is to handle the "control" flow, meaning, send the Tx to the wallet, wait for signature, catch errors and so on. You don't want to embed specific contract logic in there because it is meant to be generic, the parent component using the button is the one knowing what to do with the results and the errors. While it may seem like a convenient idea to put "business" logic in the button, you'll end up having to handle all possible contract error case which is definitely not what you want. You need to use the `onSimulateError` callback to update the label accordingly and pass it to the button

Reply:
> you'll end up having to handle all possible contract error case which is definitely not what you want

No, we will not have to do this